pkg/kudoctl/cmd/install/install.go

```diff
@@ -110,152 +65,14 @@ func installOperator(operatorArgument string, options *Options, fs afero.Fs, set
-	}
-	return nil
-}
-
-func validateCrds(crds *packages.PackageCRDs, skipInstance bool) error {
-	if skipInstance {
-		// right now we are just validating parameters on instance, if we're not creating instance right now, there is nothing to validate
-		clog.V(3).Printf("skipping instance...")
-		return nil
-	}
-	parameters := crds.OperatorVersion.Spec.Parameters
-	missingParameters := []string{}
-	for _, p := range parameters {
-		if p.Required && p.Default == nil {
-			_, ok := crds.Instance.Spec.Parameters[p.Name]
-			if !ok {
-				missingParameters = append(missingParameters, p.Name)
-			}
-		}
-	}
-
-	if len(missingParameters) > 0 {
-		return clog.Errorf("missing required parameters during installation: %s", strings.Join(missingParameters, ","))
-	}
-	return nil
-}
-
-// VersionExists looks for string version inside collection of versions
-func VersionExists(versions []string, currentVersion string) bool {
-	for _, v := range versions {
-		if v == currentVersion {
-			return true
-		}
-	}
-	return false
-}
-
-// installSingleOperatorToCluster installs a given Operator to the cluster
-// TODO: needs testing
-func installSingleOperatorToCluster(name, namespace string, o *v1alpha1.Operator, kc *kudo.Client) error {
-	if _, err := kc.InstallOperatorObjToCluster(o, namespace); err != nil {
-		return errors.Wrapf(err, "installing %s-operator.yaml", name)
-	}
-	clog.Printf("operator.%s/%s created", o.APIVersion, o.Name)
-	return nil
-}
-
-// installSingleOperatorVersionToCluster installs a given OperatorVersion to the cluster
-// TODO: needs testing
-func installSingleOperatorVersionToCluster(name, namespace string, kc *kudo.Client, ov *v1alpha1.OperatorVersion) error {
-	if _, err := kc.InstallOperatorVersionObjToCluster(ov, namespace); err != nil {
-		return errors.Wrapf(err, "installing %s-operatorversion.yaml", name)
-	}
-	clog.Printf("operatorversion.%s/%s created", ov.APIVersion, ov.Name)
-	return nil
-}
-
-// installSingleInstanceToCluster installs a given Instance to the cluster
-// TODO: needs more testing
-func installSingleInstanceToCluster(name string, instance *v1alpha1.Instance, kc *kudo.Client, options *Options, settings *env.Settings) error {
-	if _, err := kc.InstallInstanceObjToCluster(instance, settings.Namespace); err != nil {
-		return errors.Wrapf(err, "installing instance %s", name)
-	}
-	clog.Printf("instance.%s/%s created", instance.APIVersion, instance.Name)
-	return nil
-}
-
-func applyInstanceOverrides(instance *v1alpha1.Instance, options *Options) {
-	if options.InstanceName != "" {
-		instance.ObjectMeta.SetName(options.InstanceName)
-		clog.V(3).Printf("instance name: %v", options.InstanceName)
-	}
-	if options.Parameters != nil {
-		instance.Spec.Parameters = options.Parameters
-		clog.V(3).Printf("parameters in use: %v", options.Parameters)
-	}
+func installCrds(resources *packages.Resources, kc *kudo.Client, options *Options, settings *env.Settings) error {
```

Comment:
This method does not install CR**D**s. wdyt about:
```suggestion
func installPackage(resources *packages.Resources, kc *kudo.Client, options *Options, settings *env.Settings) error {
```

Reply:
but also: why do you need it at all: it's just a wrapper around `kudo.Install` so you could just call it directly above.

Reply:
That's only there to keep the tests in `install_test.go` running. But since then things changed a bit, let me see if I can move these tests to the `kudo` package and have them test `kudo.Install` instead. Same for the code in `cmd.upgrade`.